### PR TITLE
More refactor toward package unit tests - particularly around app package.

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

**Comment by nsoranzo** [comment]
Shouldn't `test/unit/dataset_collections/test_matching.py` go to test/unit/data/dataset_collections/test_matching.py` (instead of `app`) ?

File: test/unit/app/jobs/test_job_configuration.py [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
 # File would be slightly more readable if contents were embedded directly, but
 # there are advantages to testing the documentation/examples.
-SIMPLE_JOB_CONF = os.path.join(os.path.dirname(__file__), "..", "..", "..", "lib", "galaxy", "config", "sample", "job_conf.xml.sample_basic")
-ADVANCED_JOB_CONF = os.path.join(os.path.dirname(__file__), "..", "..", "..", "lib", "galaxy", "config", "sample", "job_conf.xml.sample_advanced")
+SIMPLE_JOB_CONF = os.path.join(os.path.dirname(__file__), "..", "..", "..", "..", "lib", "galaxy", "config", "sample", "job_conf.xml.sample_basic")
+ADVANCED_JOB_CONF = os.path.join(os.path.dirname(__file__), "..", "..", "..", "..", "lib", "galaxy", "config", "sample", "job_conf.xml.sample_advanced")
```
Comment (nsoranzo) [inline test/unit/app/jobs/test_job_configuration.py]:
```suggestion
SIMPLE_JOB_CONF = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, os.pardir, "lib", "galaxy", "config", "sample", "job_conf.xml.sample_basic")
ADVANCED_JOB_CONF = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, os.pardir, "lib", "galaxy", "config", "sample", "job_conf.xml.sample_advanced")
```

File: test/unit/app/jobs/test_job_configuration.py [hunk 2]
```diff
@@ -396,8 +396,8 @@ class AdvancedJobConfYamlParserTestCase(AdvancedJobConfXmlParserTestCase):
 
 
 def test_yaml_advanced_validation():
-    schema = os.path.join(os.path.dirname(__file__), "..", "..", "..", "lib", "galaxy", "webapps", "galaxy", "job_config_schema.yml")
-    integration_tests_dir = os.path.join(os.path.dirname(__file__), "..", "..", "..", "test", "integration")
+    schema = os.path.join(os.path.dirname(__file__), "..", "..", "..", "..", "lib", "galaxy", "webapps", "galaxy", "job_config_schema.yml")
+    integration_tests_dir = os.path.join(os.path.dirname(__file__), "..", "..", "..", "..", "test", "integration")
```
Comment (nsoranzo) [inline test/unit/app/jobs/test_job_configuration.py]:
```suggestion
    schema = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, os.pardir, "lib", "galaxy", "webapps", "galaxy", "job_config_schema.yml")
    integration_tests_dir = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, os.pardir, "test", "integration")
```

**Comment by nsoranzo** [comment]
I don't think this move is safe: `test/unit/test_rule_utils.py` → `test/unit/app/test_rule_utils.py` since `test_rule_utils.py` contains this import:

```python
from galaxy_test.base import rules_test_data
```

**Comment by jmchilton** [comment]
> I don't think this move is safe: test/unit/test_rule_utils.py → test/unit/app/test_rule_utils.py since test_rule_utils.py contains this import:

For sure - but also that test belongs in galaxy-util. I know my preferred fix for this and I was worried it might be controversial (move the specification for (namely ``rules_dsl_spec.yml``)  into a file packaged with galaxy-util) so I wanted to do that in a follow up PR before enabling the app unit tests. Can I do that in a follow up PR or do you want to hash that out now?

**Comment by nsoranzo** [comment]
> > I don't think this move is safe: test/unit/test_rule_utils.py → test/unit/app/test_rule_utils.py since test_rule_utils.py contains this import:
> 
> For sure - but also that test belongs in galaxy-util. I know my preferred fix for this and I was worried it might be controversial (move the specification for (namely `rules_dsl_spec.yml`) into a file packaged with galaxy-util) so I wanted to do that in a follow up PR before enabling the app unit tests. Can I do that in a follow up PR or do you want to hash that out now?

Follow-up PR is surely fine, but what about leaving `test_rule_utils.py` in `test/unit/` in the mean time?

**Comment by nsoranzo** [comment]
`test/unit/tools/test_cwl_util.py` could go into `test/unit/tool_util/` (instead of `app/tools/` )

**Comment by nsoranzo** [comment]
Can you remove `Cheetah3` from `packages/web_framework/requirements.txt` ? It's not needed any more since https://github.com/galaxyproject/galaxy/commit/488fe1971d58516a644e447dde348263f8401141 .

**Comment by jmchilton** [comment]
I addressed your comments as three more specific PRs I think, I'll close this and reopen when they are merged.

**Pull request closed without merging**